tensorflow/contrib/rnn/python/ops/rnn_cell.py

```diff
@@ -2352,52 +2332,48 @@ def call(self, inputs, state):
       for group_id in range(self._number_of_groups):
         with vs.variable_scope("group%d" % group_id):
           x_g_id = array_ops.concat(
-              [
-                  self._get_input_for_group(inputs, group_id,
-                                            self._group_shape[0]),
-                  self._get_input_for_group(m_prev, group_id,
-                                            self._group_shape[0])
-              ],
-              axis=1)
-          if self._linear1 is None:
-            self._linear1 = _Linear(x_g_id, 4 * self._group_shape[1], False)
-          R_k = self._linear1(x_g_id)  # pylint: disable=invalid-name
+            [self._get_input_for_group(inputs, group_id,
```

Comment:
Is there a necessity to name each input for a group? Otherwise all the explicit indexing could be avoided:
input_parts = tf.split(inputs, self._number_of_groups)
m_prev_parts = tf.split(m_prev, self._number_of_groups)
for group_id, (input_part, m_prev_part) in enumerate(zip(input_parts, m_prev_parts)):
...